ddtrace/internal/telemetry/writer.py

```diff
@@ -465,7 +465,11 @@ def remove_configuration(self, configuration_name):
     def add_configuration(self, configuration_name, configuration_value, origin="unknown"):
         # type: (str, Any, str) -> None
         """Creates and queues the name, origin, value of a configuration"""
-        if not isinstance(configuration_value, (bool, str, int, float, type(None))):
+        if isinstance(configuration_value, dict):
+            configuration_value = ",".join(":".join((k, str(v))) for k, v in configuration_value.items())
+        elif isinstance(configuration_value, (list, tuple)):
+            configuration_value = ",".join(str(v) for v in configuration_value)
+        elif not isinstance(configuration_value, (bool, str, int, float, type(None))):
```

Comment:
Dumb question: Why are there so many ways that configs can be represented?

Reply:
Telemetry payloads are serialized as json,  json supports seralizing the following python types: `None, int, float, str, bool, List[_JSONType], Dict[str, _JSONType]`. `configuration_value` is not one of these types (ex: a set) then we will fail to generate and send telemetry events.